internal_transfer_with_agreed_amount/wizards/internal_transfer_multicurrency.py

```diff
@@ -0,0 +1,85 @@
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+
+class InternalTransferMulticurrency(models.TransientModel):
+    _name = "internal.transfer.multicurrency"
+    _description = "Wizard to modify the agreed amount with the bank on an internal transfer."  # noqa
+
+    agreed_amount = fields.Monetary(help="Agreed amount in the currency of the destination bank.")
+    currency_id = fields.Many2one(
+        "res.currency",
+        required=True,
+        default=lambda self: self._get_currency(),
+    )
+
+    def _get_currency(self):
+        active_ids = self._context.get("active_ids")
+        active_model = self._context.get("active_model")
+        if not active_ids or active_model != "account.payment":
+            return False
+        payment = self.env["account.payment"].browse(active_ids)
+        return payment.destination_journal_id.currency_id or payment.company_currency_id
+
+    def _validate_currencies_in_internal_transfer(self, payment):
+        payment_currency = payment.currency_id
+        company_currency = payment.company_currency_id
+        send_currency = payment.journal_id.currency_id or company_currency
+        receive_currency = self.currency_id
+        currencies = payment_currency | company_currency | send_currency | receive_currency
+        if len(currencies) == 2:
+            return currencies - payment_currency
+        raise ValidationError(
+            _(
+                """
+                Agreed amount is available when there are two considered currencies.
+                Currently, the %s currencies are considered for this process.
+                """,
+                ", ".join(currencies.mapped("name")),
+            )
+        )
+
+    def _prepare_values_from_lines(self, lines, currency):
+        line = lines[0]
+        vals = {"debit": line.debit} if line.amount_currency > 0 else {"credit": line.credit}
+        sign = 1 if line.amount_currency > 0 else -1
+        vals.update({"amount_currency": sign * self.agreed_amount, "currency_id": currency.id})
+        return vals
+
+    def apply(self):
+        active_ids = self._context.get("active_ids")
+        active_model = self._context.get("active_model")
+        if not active_ids or active_model != "account.payment":
+            return
+        payment = self.env["account.payment"].browse(active_ids)
+        exchange_currency = self._validate_currencies_in_internal_transfer(payment)
+        payment.filtered(lambda p: p.state == "draft").action_post()
+        amls = (payment | payment.paired_internal_transfer_payment_id).line_ids
+        if not self.agreed_amount:
+            return
+        to_reconcile = amls.filtered(
+            lambda l: l.account_id == l.company_id.transfer_account_id
+        ).full_reconcile_id.reconciled_line_ids
```

Comment:
Are you sure relying on the full reconcile is safe?

Take into account, as its name implies, it's only created when it's fully reconciled.

Maybe a better approach would be to use partial reconciles (matching debit and matching credits)

If @hbto told you that way is safe, I'm OK with that.

Reply:
Thanks for your reply @luisg123v 

Actually there is no case in doing anything on this records if the lines of the screened account is not fully reconciled.

Please Go Ahead and God Speed.

Best Regards.

Reply:
Gread @hbto, thanks for the reply. Resolving.